rosdoc2/verbs/build/builders/sphinx_builder.py

```diff
@@ -366,10 +405,55 @@ def build(self, *, doc_build_folder, output_staging_directory):
             if package_name != self.build_context.package.name
         ]
 
+        # If the package has build type `ament_python`, or if the user configured
+        # to run `sphinx-apidoc`, then invoke `sphinx-apidoc` before building
+        if (
+            self.build_context.build_type == 'ament_python' or
+                self.build_context.always_run_sphinx_apidoc
+        ):
+            package_xml_directory = os.path.dirname(self.build_context.package.filename)
+            # If 'python_source' is specified, construct 'package_src_directory' from it
+            if self.build_context.python_source is not None:
+                package_src_directory = \
+                    os.path.join(
+                        package_xml_directory,
+                        self.build_context.python_source)
+            # If not provided, try to find the package source direcotry
+            else:
+                package_list = setuptools.find_packages(where=package_xml_directory)
+                try:
+                    package_src_directory = \
+                        os.path.join(
+                            package_xml_directory,
+                            package_list[package_list.index(self.build_context.package.name)])
+                except ValueError:
+                    package_src_directory = None
```

Comment:
Ok, I see how this address the "over matching" of `setuptools.find_packages()` 👍.

However, the try-except is kind of odd. I think it's ok to leave as-is, but an explicit `if ... in package_list` might be easier to understand. At first glance it's not clear where the `ValueError` is coming from. Maybe something like:

```suggestion
                if self.build_context.package.name in package_list:
                    package_src_directory = os.path.join(
                        package_xml_directory,
                        self.build_context.package.name)
                else:
                    package_src_directory = None
```